.github/workflows/main.yml

```diff
@@ -2,13 +2,11 @@ name: Build-Android
 
 on:
   push:
-    branches: [ "main", "frontend" ]
+    branches: [ "main"]
     paths: 
       - "app/**"
   pull_request:
-    branches: [ "main", "frontend" ]
-    paths: 
-      - "app/**"
+    branches: [ "main"]
   workflow_dispatch:
   
 jobs:
```

Comment:
we should rename the build jobt to maybe `build_flutter` because it currently is ambiguous with the secondary build job required for build_rust. On the merge requirements, I can only specify a name of a job. Currently, when specifying build, the backend build job is required redundantly (as it is also a dependency of build_rust)